src/dsp_tools/utils/xml_validation_models.py

```diff
@@ -0,0 +1,143 @@
+        <text permissions="prop-default" encoding="xml">Text 2</text>
+    </text-prop>
+    ```
+
+    Args:
+        root: root of the data xml document
+
+    Returns:
+          True and None if all the elements are consistent
+          False and a list of all the inconsistent <text-props>
+    """
+    text_props = _get_all_ids_prop_encoding_from_root(root)
+    if (checked_props := _check_only_one_valid_encoding_used_all_props(text_props)) is not None:
+        return False, checked_props
+    return True, None
+
+
+def _get_all_ids_prop_encoding_from_root(
+    root: etree._Element,
+) -> list[TextValueData]:
+    res_list = []
+    for res_input in root.iterchildren(tag="resource"):
+        if (res_result := _get_id_prop_encoding_from_one_resource(res_input)) is not None:
+            res_list.extend(res_result)
+    return res_list
+
+
+def _get_id_prop_encoding_from_one_resource(resource: etree._Element) -> list[TextValueData] | None:
+    if not (children := list(resource.iterchildren(tag="text-prop"))):
+        return None
+    res_id = resource.attrib["id"]
+    return [_get_prop_encoding_from_one_property(res_id, child) for child in children]
+
+
+def _get_prop_encoding_from_one_property(res_id: str, property: etree._Element) -> TextValueData:
+    prop_name = property.attrib["name"]
+    child_attrib = [x.attrib for x in property.iterchildren()]
+    encodings = {x["encoding"] for x in child_attrib}
+    return TextValueData(res_id, prop_name, encodings)
+
+
+def _check_only_one_valid_encoding_used_all_props(text_props: list[TextValueData]) -> list[TextValueData] | None:
+    wrong_props = [x for x in text_props if not _check_only_one_valid_encoding_used_one_prop(x.encoding)]
+    if len(wrong_props) == 0:
+        return None
+    return wrong_props
+
+
+def _check_only_one_valid_encoding_used_one_prop(text_encodings: set[str]) -> bool:
+    if text_encodings == {"xml"} or text_encodings == {"utf8"}:  # noqa: PLR1714
+        return True
+    return False
+
+
+@dataclass
+class InconsistentTextValueEncodings:
+    """
+    This class takes instances that contain the information about resources and the properties that contain
+    invalid encodings.
+
+    An invalid encoding would be a <text-prop> element, that contains
+    <text encoding="utf8">
+    and
+    <text encoding="xml">
+    It is responsible to communicate the problems to the user.
+    """
+
+    problematic_resources: list[TextValueData]
+
+    def execute_problem_protocol(self) -> tuple[str, pd.DataFrame | None]:
+        """
+        This method composes an error message for the user.
+
+        Returns:
+            the error message and a dataframe with the errors if they exceed 50
+        """
+        msg = (
+            "\nSome <text-prop> elements contain <text> elements that use both 'xml' and 'utf8' encoding.\n"
+            "Only one encoding type can be used per <text-prop> element."
+        )
```

Comment:
The order how you do things in this method seems very unnatural to me. You do:

- compose a base message
- create a dataframe
- if the dataframe is big enough to be saved --> return it together with the message
    - on the caller side, you add the info that the file was saved. For me, this would belong here.
- compose an additional message from the dataframe 
    - here it gets really confusing: at that point, the dataframe will not be used, but you use it to create an "additional" message 
    - it is not clear what "additional" means in this context
- return the additional message.

My suggestion for a more natural flow of things: 

- compose a base message
- based on `problematic_resources`, determine whether to save a df or not
- if not:
    - based on `problematic_resources`, compose a print message
    - return it
- if yes:
    - create a dataframe
    - to the base message, add that a dataframe was saved, and where it was saved
    - return df + message

Reply:
Where the df is saved:
- I did it this way because in the past that is how we structured it. The rationale if I remember correctly is for better testing.
- Another reason for the location of the save message is, that it is in fact not saved in that function, therefore the message would be technically wrong in my opinion.

Composing the df:
- The reason why, I did it first, is because for a clean message, I want to group by resource id and then sort by prop name. This can both be done nicely in a df, since we may in any case create a df, I did not see a reason not to do it this way. In my opinion any other method I can think of that creates the same result is more complicated, so I don't think we should change this.

Additional msg:
- I will restructure as suggested below, then the problem with the variable name is solved.